pkg/controller/usersignup/usersignup_controller.go

```diff
@@ -97,5 +119,253 @@ func (r *ReconcileUserSignup) Reconcile(request reconcile.Request) (reconcile.Re
 		return reconcile.Result{}, err
 	}
 
+	// Check if the MasterUserRecord already exists
+	mur := &toolchainv1alpha1.MasterUserRecord{}
+	// We can use the same NamespacedName from the request as the MasterUserRecord will have the same name as the
+	// UserSignup (i.e. the user's username)
+	err = r.client.Get(context.TODO(), request.NamespacedName, mur)
+	if err != nil {
+		// We generally EXPECT the MasterUserRecord to not be found here, so we only deal with other error types
+		if !errors.IsNotFound(err) {
+			// Error reading the MasterUserRecord, requeue the request
+			return reconcile.Result{}, err
+		}
+	} else {
+		// If we successfully found an existing MasterUserRecord then our work here is done, set the status
+		// to Complete and return
+		statusError := r.setStatusComplete(instance, "")
+		if statusError != nil {
+			reqLogger.Error(statusError, "Error updating UserSignup Status to Complete", "Name", instance.Name)
+			return reconcile.Result{}, statusError
+		}
+		return reconcile.Result{}, nil
+	}
+
+	// Check the user approval policy.
+	userApprovalPolicy, err := r.ReadUserApprovalPolicyConfig()
+	if err != nil {
+		statusError := r.setStatusFailedToReadUserApprovalPolicy(instance, err.Error())
+		if statusError != nil {
+			reqLogger.Error(statusError, "Error updating UserSignup Status", "Name", instance.Name)
+			return reconcile.Result{}, statusError
+		}
+
+		reqLogger.Error(err, "Error reading user approval policy")
+		return reconcile.Result{}, err
+	}
+
+	// If the signup has been explicitly approved (by an admin), or the user approval policy is set to automatic,
+	// then proceed with the signup
+	if instance.Spec.Approved || userApprovalPolicy == config.UserApprovalPolicyAutomatic {
+		if instance.Spec.Approved {
+			statusError := r.setStatusApprovedByAdmin(instance, "")
+			if statusError != nil {
+				reqLogger.Error(statusError, "Error updating UserSignup Status", "Name", instance.Name)
+				return reconcile.Result{}, statusError
+			}
+		}
+
+		if userApprovalPolicy == config.UserApprovalPolicyAutomatic {
```

Comment:
Could be just `} else {` in line 166 to avoid two status updates (ApprovedByAdmin and then right after: ApprovedAutomaticaly) in case when both spec.Approved == true and userApprovalPolicy == config.UserApprovalPolicyAutomatic.
Or vice versa if we want Automatic approval status always win.

Reply:
Makes sense, fixed.